tests/tcp_udp_listener/tcp_udp_listener.cpp

```diff
@@ -33,9 +38,17 @@ create_listener(_Inout_ receiver_socket_t* receiver_socket)
     receiver_socket->complete_async_receive(WSA_INFINITE, false);
     printf("Received data from remote\n");
 
-    // Send a response back.
-    std::string response = SERVER_MESSAGE + std::to_string(_local_port);
+    // Query for the redirect record.
+    // This is expected to only be valid for local redirections.
+    // If not present, use the generic SERVER_MESSAGE response.
+    if (receiver_socket->query_redirect_context(
+            &redirect_context_buffer, REDIRECT_CONTEXT_BUFFER_SIZE, redirect_context_size)) {
+        response = SERVER_MESSAGE + std::to_string(_local_port);
+    } else {
+        response = redirect_context_buffer + std::to_string(_local_port);
```

Comment:
From above, the redirect_context_buffer is empty. Can it be pre-populated, to give a better debugging?

Reply:
Can you help me understand what cases you are thinking about for debugging here?

In our test cases, this listener will respond back to the client with a message, which it checks and may fail the test if the unexpected result is received. For remote redirection, SERVER_MESSAGE is expected. For local redirection, we now expect the redirect_context to be returned.

In cases where redirect_context_buffer remains empty but redirect_context was expected, we would fail (which is expected).

Reply:
Suggestion: For local redirect, in the case where redirect_context_buffer is empty:
redirect_context_buffer[] = "Local redirect failed. " 
+ Print the last error via WSAGetLastError()

Reply:
To send with the error reason (maybe get it from WSAGetLastError)